app/controllers/idv/address_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Idv
+  class AddressController < ApplicationController
+    include IdvSession
+    include IdvFailureConcern
+
+    attr_reader :idv_form
+
+    before_action :confirm_two_factor_authenticated
+
+    def new
+      analytics.track_event(Analytics::IDV_ADDRESS_VISIT)
+      set_idv_form
+      render 'idv/address'
+    end
+
+    def update
+      set_idv_form
+      form_result = idv_form.submit(profile_params)
+      analytics.track_event(Analytics::IDV_ADDRESS_SUBMITTED, form_result.to_h)
+      if form_result.success?
+        success
+      else
+        failure
+      end
+    end
+
+    private
+
+    def success
+      profile_params.each do |key|
+        user_session['idv/doc_auth']['pii_from_doc'][key] = profile_params[key]
+      end
+      redirect_to idv_doc_auth_url
+    end
+
+    def failure
+      redirect_to idv_address_url
+    end
+
+    def set_idv_form
```

Comment:
Related to not need to call `set_idv_form` above, if this isn't gonna be in a before action, and you don't need the ivar in the new view, then couldn't you rename this to `idv_form` and use it where it is needed?